### PR TITLE
Update TMA script for new sfall features.

### PR DESCRIPTION
**Pull request by FakelsHub**

### CODE REVIEW

File: Fallout2/Fallout1in2/Mapper/source/scripts/GlobalScripts/gl_tellmeabout.ssl [hunk 1]
```diff
@@ -159,14 +140,15 @@ procedure GameModeChange_Handler begin
...
+      // TODO: !!! with that all is well !!!
       // This gets called when using dialog to enter the barter interface.
       // So after using the barter dialog option, TMAs aren't working anymore
       // until the player has restarted the dialog.
       tma_data_array[0] := 0; // Unset TMA data when exiting dialog mode
       tma_gvar_array[0] := 0;
+      debug_msg("Unset TMA data.");
    end
-   else if (tellme_button and (game_mode bwand BARTER or game_mode bwand SPECIAL)) then begin
+   else if (tellme_button and (game_mode bwand SPECIAL or game_mode bwand BARTER)) then begin
```
Comment (FakelsHub) [inline Fallout2/Fallout1in2/Mapper/source/scripts/GlobalScripts/gl_tellmeabout.ssl]: add/replace `game_mode bwand SPECIAL` to `game_mode bwand (DIALOG + SPECIAL) `

**Comment by FakelsHub** [comment]
> Based on debug message, the code gets called the moment I press the dialog option, thus the array gets cleared. This does not happen if I use the barter button.

Strange a script behavior.

File: Fallout2/Fallout1in2/Mapper/source/scripts/GlobalScripts/gl_tellmeabout.ssl (hunk 1 above)
Comment (Lexx2k) [reply Fallout2/Fallout1in2/Mapper/source/scripts/GlobalScripts/gl_tellmeabout.ssl]: Ok.

PS: I tried adding in the old dialog_obj check to see if I can fix my bug that way, but it doesn't work anymore either. For some reason, as soon as I click the dialog option to enter barter screen, the game thinks I have exited the dialog and "Unset TMA data." appears in debug log.

**Comment by FakelsHub** [comment]
> PS: I tried adding in the old dialog_obj check to see if I can fix my bug that way, but it doesn't work anymore either. For some reason, as soon as I click the dialog option to enter barter screen, the game thinks I have exited the dialog and "Unset TMA data." appears in debug log.

send me your ddraw.ini

**Comment by Lexx2k** [comment]
[Here](https://pastebin.com/KtSFWgkS).

File: Fallout2/Fallout1in2/Mapper/source/scripts/headers/debug.h [hunk 2]
```diff
@@ -6,4 +6,11 @@
 #define debug_mode  (global_var(GVAR_ENABLE_DEBUG_MESSAGES) == 1)
 #define debug(x)    if debug_mode then display_msg(x)
 
+// Macros for debugging messages
+#ifdef DEBUG
+#define DEBUGMSG(x)       debug_msg(x);
+#else
+#define DEBUGMSG(x)       //nothing
+#endif
```
Comment (wipe2238) [inline Fallout2/Fallout1in2/Mapper/source/scripts/headers/debug.h]: This has to go; nothing worse than N ways to display debug spam, each working differently,

Comment (FakelsHub) [reply Fallout2/Fallout1in2/Mapper/source/scripts/headers/debug.h]: What?

**Comment by FakelsHub** [comment]
> Here.

this is no different from mine: /

File: Fallout2/Fallout1in2/Mapper/source/scripts/headers/debug.h (hunk 2 above)
Comment (Lexx2k) [reply Fallout2/Fallout1in2/Mapper/source/scripts/headers/debug.h]: Wipe means, it is a #define of a #define of a #define for a gvar condition. :>

Comment (wipe2238) [reply Fallout2/Fallout1in2/Mapper/source/scripts/headers/debug.h]: I mean we have debug() which can be switched on and off via .ini, there's no point introducing another basing on defines.

Comment (Lexx2k) [reply Fallout2/Fallout1in2/Mapper/source/scripts/headers/debug.h]: debug(x) prints text to ingame message log, though while debug_msg(x) prints to debug.log :p

Comment (FakelsHub) [reply Fallout2/Fallout1in2/Mapper/source/scripts/headers/debug.h]: `DEBUGMSG ()` - this based on conditional compilation, macro is set via compiler option [-mDEBUG](https://github.com/FakelsHub/int-sslc/blob/b97c184d09e4bd6285b9c5da4701fedb399483f6/scr/compile.c#L71)

Comment (wipe2238) [reply Fallout2/Fallout1in2/Mapper/source/scripts/headers/debug.h]: I know that, i just see no point introducing such way to turn on _part of_ debug messages.
Change .ini for this spam, compile with -m for other spam, bleh :/

**Pull request merged**